meetings.py

```diff
@@ -10,80 +12,94 @@
 from dotenv import load_dotenv
 load_dotenv(join(dirname(__file__), '.env'))
 
+KEY = getenv('ZOOM_KEY')
+SECRET = getenv('ZOOM_SECRET')
+API_BASE_URL = "https://api.zoom.us/v1"
+MEETING_TYPES = {"live": 1, "past": 2}
 
-def main():
-    parser = argparse.ArgumentParser()
-    parser.add_argument("--key", help="zoom api key")
-    parser.add_argument("--secret", help="zoom api secret")
-    parser.add_argument("--destination", help="destination filename")
-    args = parser.parse_args()
-
-    if args.key is None:
-        KEY = getenv('ZOOM_KEY')
+class ZoomApiException(Exception):
+    pass
 
-    if args.secret is None:
-        SECRET = getenv('ZOOM_SECRET')
+def get_meetings_from(date, type, key, secret):
 
-    FILENAME = args.destination
-    MEETINGS = "https://api.zoom.us/v1/metrics/meetings"
-
-    defaults = {
-        'api_key': KEY,
-        'api_secret': SECRET,
-        'type': "2",
-        'page_number': 1,
-        'page_size': "300",
+    params = {
+        'from': date,
+        'to': date,
+        'type': MEETING_TYPES[type],
+        'api_key': key,
+        'api_secret': secret,
+        'page_size': 100,
+        'page_number': 1
     }
 
-    meeting_types = {
-        'live': 1,
-        'past': 2
-    }
-
-    destination = open(FILENAME, "w")
-
-    def get_meetings_from(date, type):
-        params = defaults.copy()
-        params['from'] = date
-        params['to'] = date
-        params['type'] = meeting_types[type]
-
-        while True:
-            r = requests.post(url=MEETINGS, data=params)
-
-            if r.status_code != requests.codes.ok:
-                print("Failed with status code:", r.status_code, "reason:", r.reason)
-                return
+    meetings = []
+    while True:
+
+        r = requests.post(url=API_BASE_URL + "/metrics/meetings", data=params)
+        r.raise_for_status()
+        response = r.json()
+
+        if 'error' in response.keys():
+            raise ZoomApiException(response['error'])
+
+        for meeting in response['meetings']:
+            meeting['type'] = response['type']
```

Comment:
i think this wasn't working but didn't look too closely. the meeting data seemed to always have a "type" value like "Past Meetings" even after modifying it here.